src/Host.cpp

```diff
@@ -1921,6 +1921,12 @@ bool Host::uninstallPackage(const QString& packageName, int module)
     //     0=package, 1=uninstall from dialog, 2=uninstall due to module syncing,
     //     3=uninstall from a script
 
+    // block packages/modules from being uninstalled while a profile save is in progress
+    // just so the save mechanism doesn't get surprised with something getting removed from memory under its feet
+    if (currentlySavingProfile()) {
+        return false;
```

Comment:
Although we return a `false` on failure, this method is called from, I think around ten places - but only one of them (the one in `(int) TLuaInterpreter::uninstallModule(lua_State*)`) actually bothers to take notice of the returned value. As such I think we are going to get further borkage from at least some of the other places because we are not considering that a retry will be needed to actually get the package/module removed. This isn't intended as a criticism of this PR but I want to raise it as a concern. :worried: